### PR TITLE
Initial website setup

### PR DESCRIPTION
**Pull request by cskartikey**
- Created project structure with necessary directories and files
- Set up Next.js with Tailwind CSS and Font Awesome
- Added base HTML structure and layout components
- Styled the homepage with basic styling
- Added FontAwesome icons
- Configured font imports and styles
- Integrated HackClub branding elements

This commit establishes the foundation for our website, including the project structure, styling, and initial content.

**To-Dos**

- [x] Rebrand it with HC theme https://theme.hackclub.com/
- [x] Add the split button
- [ ] A design mockup for docs

### CODE REVIEW

**Comment by devramsean0** [comment]
Could we move the Google Fonts requests locally? I'm cautious of the Privacy implications of calling any google services.
This guide could help and it would be a really quick change https://fonts.google.com/knowledge/using_type/self_hosting_web_fonts

**Comment by polypixeldev** [comment]
Since it's NextJS, you can use their automatic font optimization: https://nextjs.org/docs/app/building-your-application/optimizing/fonts (which can also integrate with tailwind)

**Comment by polypixeldev** [comment]
I think HackClub should have a space so that it's Hack Club, but that's all - looks great!

**Comment by cskartikey** [comment]
> I think HackClub should have a space so that it's Hack Club, but that's all - looks great!

Can you elaborate on this?

File: site/pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                        <span className="text-burrowPurple font-bold">Burrow</span> Through{" "}
+                        <span className="font-SpaceMono font-bold">Firewall</span>
+                    </h1>
+                    <div className="text-lg md:text-2xl font-Poppins my-2 md:my-10 w-11/12 md:w-3/4 mx-auto">
+                        <p>
+                            Burrow is an open source tool for burrowing through firewalls,
+                            built by teenagers at{" "}
+                            <span className="font-PhantomSans text-hackClubRed underline">
+                                <a href="https://www.hackclub.com/" target="_blank">
+                                    HackClub.
```
Comment (polypixeldev) [inline site/pages/index.tsx:23]:
```suggestion
                                    Hack Club.
```

File: site/pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                        <a href="https://hackclub.com/" target="_blank">
+                            <button className="flex items-center bg-transparent border-2 border-burrowStroke text-hackClubRed font-SpaceMono text-lg md:text-2xl rounded-xl md:rounded-2xl h-12 md:h-16 px-3">
+                                <Image
+                                    src="/hackclub.svg"
+                                    width={35}
+                                    height={35}
+                                    className="mx-2"
+                                    alt="HackClub's logo"
+                                />
+                                By HackClub
```
Comment (polypixeldev) [inline site/pages/index.tsx:70]:
```suggestion
                                By Hack Club
```

**Review by polypixeldev — DISMISSED** [review]
Like this:

File: site/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+- [Next.js Documentation](https://nextjs.org/docs) - learn about Next.js features and API.
+- [Learn Next.js](https://nextjs.org/learn) - an interactive Next.js tutorial.
+
+You can check out [the Next.js GitHub repository](https://github.com/vercel/next.js/) - your feedback and contributions are welcome!
+
+## Deploy on Vercel
+
+The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.
+
+Check out our [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.
```
Comment (Muirrum) [inline site/README.md:34]: Ideally this will be more descriptive, but I won't make it a blocker for the merge

File: site/pages/index.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            document.removeEventListener('click', handleClickOutside);
+        };
+    }, []);
+    return (
+        <>
+            <Head>
+                <title>Burrow</title>
+                <meta
+                    name="description"
+                    content="Burrow is an open-source tool for bypassing firewalls, built by teenagers at HackClub."
```
Comment (Muirrum) [inline site/pages/index.tsx:35]:
```suggestion
                    content="Burrow is an open-source tool for bypassing firewalls, built by teenagers at Hack Club."
```

File: site/pages/index.tsx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                                    src="/hackclub.svg"
+                                    width={35}
+                                    height={35}
+                                    className="mx-2"
+                                    alt="Hack Club's logo"
+                                />
+                                By Hack Club
+                            </button>
+                        </a>
+                    </div> */}
```
Comment (Muirrum) [inline site/pages/index.tsx:151]: Can you give some context as to why this is commented out?

File: site/pages/index.tsx [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+                            <span className="text-hackClubRed underline">
+                                <a href="https://www.hackclub.com/" target="_blank">
+                                    Hack Club.
+                                </a>
+                            </span>{" "}
+                            <span className="rounded-md bg-burrowHover p-0.5 font-SpaceMono text-hackClubBlue">
+                                burrow
+                            </span>{" "}
+                            provides a simple command-line tool to open virtual interfaces and
+                            direct traffic through them.
```
Comment (Muirrum) [inline site/pages/index.tsx:58]: I don't think the second sentence adds to the first. Potential second sentence could be 

```suggestion
                            Burrow is an open source tool for burrowing through firewalls,
                            built by teenagers at{" "}
                            <span className="text-hackClubRed underline">
                                <a href="https://www.hackclub.com/" target="_blank">
                                    Hack Club.
                                </a>
                            </span>{" "}
                            <span className="rounded-md bg-burrowHover p-0.5 font-SpaceMono text-hackClubBlue">burrow</span>{" "} is a Rust-based VPN for getting around restrictive Internet censors.
```

I'm not attached to it though, so please feel free to suggest alternatives!

**Review by Muirrum — CHANGES_REQUESTED** [review]
Overall looks good, some minor issues with the copy

File: site/pages/index.tsx (hunk 4 above)
Comment (cskartikey) [reply site/pages/index.tsx:35]: Whoops! missed it.

File: site/pages/index.tsx (hunk 6 above)
Comment (cskartikey) [reply site/pages/index.tsx:58]: Works! I just copied it from GitHub

File: site/README.md (hunk 3 above)
Comment (cskartikey) [reply site/README.md:34]: I'll update the README in the subsequent commits. Probably after I add the docs.

Comment (cskartikey) [reply site/README.md:34]: Will make it more descriptive in the subsequent commits.

File: site/pages/index.tsx (hunk 5 above)
Comment (Muirrum) [reply site/pages/index.tsx:147]: Sounds good, thanks!

**Review by Muirrum — COMMENTED** [review]
Looks good to me! @conradev any final thoughts?

**Comment by cskartikey** [comment]
> Looks good to me! @conradev any final thoughts?

PS: fixed huge icons on load and CLS. Performance is 100% on lighthouse now.

**Review by conradev — APPROVED** [review]

**Review by Muirrum — APPROVED** [review]
LGTM! What's the plan for deploying this?

**Review by polypixeldev — CHANGES_REQUESTED** [review]
Just a small nitpick: the root package.json and package-lock.json should be deleted, as they contain an unnecessary dependency

Other than that, I love it - design is great!

**Comment by Muirrum** [comment]
Good catch! Once those get deleted, this should be good to merge.

**Pull request closed without merging**